xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -0,0 +1,148 @@
+package info.bitrich.xchangestream.kraken;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.kraken.dto.enums.KrakenEventType;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionConfig;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionMessage;
+import info.bitrich.xchangestream.kraken.dto.enums.KrakenSubscriptionName;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionStatusMessage;
+import info.bitrich.xchangestream.kraken.dto.KrakenSystemStatus;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static info.bitrich.xchangestream.kraken.dto.enums.KrakenEventType.subscribe;
+
+/**
+ * @author makarid, pchertalev
+ */
+public class KrakenStreamingService extends JsonNettyStreamingService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KrakenStreamingService.class);
+    private static final String EVENT = "event";
+    private final Map<Integer, String> channels = new HashMap<>();
```

Comment:
Thread safety. This should almost certainly be a concurrent map and use atomic mutator methods (`putIfAbsent`, `compute` etc)

Reply:
"channels" map defines mapping kraken channelID to bitrich channel name - key is generated by kraken and is unique for concrete channel name. There is impossible to subscribe the same channelName twice in NettyStreaminService (see info/bitrich/xchangestream/service/netty/NettyStreamingService.java:316). 
So we don't need care about threade safe here as well.

Reply:
See above - all you have to have is more than one thread accessing a hashmap at the same time, _even if they don't access the same keys_.

However, I've honestly not checked if that's likely in this case. If you can confirm that won't happen, I'll be happy.

Reply:
Ok. I have changed Map implemetation to ConcurrentHashMap

Reply:
Can't see this change yet.

Reply:
Sorry I did not commit one. No it is committed. Please check.